pkg/noderesourcetopology/plugin.go

```diff
@@ -120,6 +121,10 @@ func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error)
 		return nil, fmt.Errorf("want args to be of type NodeResourceTopologyMatchArgs, got %T", args)
 	}
 
+	if err := validation.ValidateNodeResourceTopologyMatchArgs(nil, tcfg); err != nil {
```

Comment:
passing `nil` here is suspiscious, is this correct? if so worth 1-line comment to remind/explain why

Reply:
This is on purpose, I have based my validation work on how in-tree plugins do it 😄 example https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L71

Reply:
nice, TIL. I'll have another, deeper, last pass later as my final review.

Reply:
Note to self: passing `nil` here is handled by `k/k/apimachinery/pkg/util/validation/field` Path objects and means "root" of the path to explore. A bit unintuitive maybe, but totally supported and idiomatic indeed.